com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -114,167 +147,494 @@ internal bool HasPositionChange
                 {
-                    serializer.SerializeValue(ref ScaleY);
+                    if (QuaternionSync)
+                    {
+                        if (IsTeleportingNextFrame)
+                        {
+                            serializer.SerializeValue(ref Rotation);
+                        }
+                        else
+                        {
+                            if (QuaternionCompression)
+                            {
+                                if (isWriting)
+                                {
+                                    QuaternionCompressed = QuaternionCompressor.CompressQuaternion(ref Rotation);
+                                }
+
+                                serializer.SerializeValue(ref QuaternionCompressed);
+
+                                if (!isWriting)
+                                {
+                                    QuaternionCompressor.DecompressQuaternion(ref Rotation, QuaternionCompressed);
+                                }
+                            }
+                            else
+                            {
+                                if (UseHalfFloatPrecision)
+                                {
+                                    if (isWriting)
+                                    {
+                                        HalfVectorRotation.FromQuaternion(ref Rotation);
+                                    }
+
+                                    serializer.SerializeNetworkSerializable(ref HalfVectorRotation);
+
+                                    if (!isWriting)
+                                    {
+                                        HalfVectorRotation.ToQuaternion(ref Rotation);
+                                    }
+                                }
+                                else
+                                {
+                                    serializer.SerializeValue(ref Rotation);
+                                }
+                            }
+                        }
+                    }
+                    else // Legacy Rotation Synchronization
+                    {
+                        // RotAngle Values
+                        if (HasRotAngleX)
+                        {
+                            serializer.SerializeValue(ref RotAngleX);
+                        }
+
+                        if (HasRotAngleY)
+                        {
+                            serializer.SerializeValue(ref RotAngleY);
+                        }
+
+                        if (HasRotAngleZ)
+                        {
+                            serializer.SerializeValue(ref RotAngleZ);
+                        }
+                    }
                 }
 
-                if (HasScaleZ)
+                // Synchronize Scale
+                if (HasScaleChange)
                 {
-                    serializer.SerializeValue(ref ScaleZ);
+                    if (UseHalfFloatPrecision)
```

Comment:
Yeah, I thought about this. If you review the NetworkTransform System Analysis document one more time you will see that when we implement the state processors this kind of change will be easier to implement.  Which that is another thing to contemplate... whether we want to use this PR as the base for that update and then just roll out all of the updates in one PR. On the other hand, this PR resolves several issues... so I have been on the fence about whether it would be better to have this as the "last update for NetworkTransform using this approach"... the biggest struggle I have been having with this PR is making the NetworkTransformState public...that makes it a long term commitment...
_(worth thinking about)_

Reply:
implemented.  👍